consensus/src/consensus_agent/requests.rs

```diff
@@ -106,4 +110,44 @@ impl<B: AbstractBlockchain<'static> + 'static, MA: MessageAdapter<B::Block> + 's
         });
         tokio::spawn(future);
     }
+
+    pub(super) fn on_get_epoch_transactions(&self, get_epoch_transactions_message: GetEpochTransactionsMessage) {
+        trace!("[GET-EPOCH-TRANSACTIONS] from {}", self.peer.peer_address());
+        if !self.state.write().epoch_transactions_limit.note_single() {
+            warn!("Rejecting GetEpochTransactions message - rate-limit exceeded");
+            // TODO: Should we send something?
+            return;
+        }
+
+        let transactions: Option<Vec<Transaction>> = self.blockchain.get_epoch_transactions(get_epoch_transactions_message.epoch, |tx| tx.clone(), None);
+        if transactions.is_none() {
+            debug!("[GET-EPOCH-TRANSACTIONS] Could not determine transactions for epoch {:?}", get_epoch_transactions_message.epoch);
+            return;
+        }
+        let transactions = transactions.unwrap();
+        let hashes: Vec<Blake2bHash> = transactions.iter().map(|tx| tx.hash()).collect();
+
+        // Fast integer division ceiling, we want ceil(#txs / MAX_TRANSACTIONS).
+        let num_chunks = (transactions.len() + EpochTransactionsMessage::MAX_TRANSACTIONS - 1) / EpochTransactionsMessage::MAX_TRANSACTIONS;
```

Comment:
Would be useful to put this into `nimiq_utils`. There is a `log2` module in there. We could rename it to `math` and put random functions for calculations in therel

Reply:
Cool idea, we use it quite often